pandas/tests/io/test_sql.py

```diff
@@ -548,9 +549,15 @@ def sqlite_buildin_iris(sqlite_buildin, iris_path):
 def test_dataframe_to_sql(conn, test_frame1, request):
     # GH 51086 if conn is sqlite_engine
     conn = request.getfixturevalue(conn)
-    test_frame1.to_sql("test", conn, if_exists="append", index=False)
+    msg = (
+        "Starting with pandas version 3.0 all arguments of to_sql except for the "
+        "argument 'name' will be keyword-only."
+    )
+    with tm.assert_produces_warning(FutureWarning, match=re.escape(msg)):
+        test_frame1.to_sql("test", conn, if_exists="append", index=False)
 
 
+@pytest.mark.skip
```

Comment:
This shouldn't be skipped

Reply:
@mroeschke I meant to include this in my comment from the other day: For this test and the one below I ran into an issue where I could not key on both FutureWarning and UserWarning. `tm.assert_produces_warning` would only catch one of the warnings, regardless of the order that I checked them in.